File: CS/06_Annotations/CreatePdfLinkAnnotation.cs
using Spire.Pdf;
using Spire.Pdf.Annotations;
using Spire.Pdf.Graphics;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace CreatePdfLinkAnnotation
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //Create a new object of PdfDocument.
            PdfDocument doc = new PdfDocument();

            //Add a page to it.
            PdfPageBase page = doc.Pages.Add();

            //Declare two parameters that will be passed to the constructor of PdfFileLinkAnnotation class.
            RectangleF rect = new RectangleF(0, 40, 250, 35);
            string filePath = @"..\..\..\..\..\..\Data\Template_Pdf_3.pdf";

            //Create a file link annotation based on the two parameters and add the annotation to the new page.
            PdfFileLinkAnnotation link = new PdfFileLinkAnnotation(rect, filePath);
            page.AnnotationsWidget.Add(link);

            //Create a free text annotation based on the same RectangleF, specifying the content.
            PdfFreeTextAnnotation text = new PdfFreeTextAnnotation(rect);
            text.Text = "Click here! This is a link annotation.";
            PdfFont font = new PdfFont(PdfFontFamily.Helvetica, 15);
            text.Font = font;
            page.AnnotationsWidget.Add(text);

            String result = "CreatePdfLinkAnnotation_out.pdf";

            //Save the document
            doc.SaveToFile(result);
            //Launch the Pdf file
            PDFDocumentViewer(result);
        }
        private void PDFDocumentViewer(string filename)
        {
            try
            {
                System.Diagnostics.Process.Start(filename);
            }
            catch { }
        }
    }
}

File: VB/06_Annotations/CreatePdfLinkAnnotation.vb
Imports Spire.Pdf
Imports Spire.Pdf.Annotations
Imports Spire.Pdf.Graphics

Namespace CreatePdfLinkAnnotation
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Create a new PDF document
            Dim doc As New PdfDocument()

            ' Add a new page to the document
            Dim page As PdfPageBase = doc.Pages.Add()

            ' Define the rectangle for the link annotation
            Dim rect As New RectangleF(0, 40, 250, 35)

            ' Specify the file path of the linked PDF file
            Dim filePath As String = "..\..\..\..\..\..\Data\Template_Pdf_3.pdf"

            ' Create a new file link annotation with the specified rectangle and file path
            Dim link As New PdfFileLinkAnnotation(rect, filePath)

            ' Add the file link annotation to the page
            page.Annotations.Add(link)

            ' Create a new free text annotation with the specified rectangle
            Dim text As New PdfFreeTextAnnotation(rect)

            ' Set the text content for the free text annotation
            text.Text = "Click here! This is a link annotation."

            ' Set the font for the free text annotation
            Dim font As New PdfFont(PdfFontFamily.Helvetica, 15)
            text.Font = font

            ' Add the free text annotation to the page
            page.AnnotationsWidget.Add(text)

            ' Save the document to a file
            Dim result As String = "CreatePdfLinkAnnotation_out.pdf"
            doc.SaveToFile(result)

            ' Close the document
            doc.Close()

            ' Launch the Pdf file
            PDFDocumentViewer(result)
        End Sub
        Private Sub PDFDocumentViewer(ByVal filename As String)
            Try
                Process.Start(filename)
            Catch
            End Try
        End Sub
    End Class
End Namespace
